### DISCUSSION TITLE
Help Installing - Bit Torrent Search Not Working

### DISCUSSION THREAD

**Discussion by ManMallard** [shiningw/ncdownloader#31] [Q&A]
I'm very new to the world of nextcloud/server stuff. I'm struggling to install this and getting it to work. 

I'm using archlinux.

I have nextcloud installed and running. 

I've installed aria2c via snap.

I installed ncdownloader via app install on nextcloud. 

When I use the app, the search works and displays files. But when I attempt to download them it just says download failed. 

Is there somewhere, or someone that can help walk me through this? Where to look for better failure information?

**Comment by shiningw** [comment]
Hello,
 which version of ncdownloader are you using? no errors displayed when going to the home page of ncdownloader? maybe you can post a screenshot or check if there are any error messages in the log

**Reply by ManMallard** [reply]
I am using ncdownloader 0.7.5.

Another issue is that when I click on the "active download" or "failed download" nothing populates the screen. When I download I get "success" but then nothing downloads and the "failed download" counter increases by one.  

Error log is the "aria2c" log. I get the same error when I try a search torrent or a youtube download. 

I had to install python, although i already have python3, as I was getting an error about that. And you'll see early in the logs I had an error where aria was already started and ncdownloader was trying to start a second copy - I think - and the port was already taken. 

My guess is that I don't have RPC configured properly? I don't know what that means or how to do it right, though. 



![image](https://user-images.githubusercontent.com/76194382/155350165-da2c8370-f392-46ef-872c-d20779e45bcf.png)

![image](https://user-images.githubusercontent.com/76194382/155350462-a885e4e4-8ac6-4a97-a763-2de1e2afcaf2.png)

`2021-11-11 07:37:56.510267 [ERROR] [DHTAutoSaveCommand.cc:112] Exception caught while saving DHT routing table to /var/www/.cache/aria2/dht.dat
Exception: [DHTRoutingTableSerializer.cc:87] errorCode=1 Failed to save DHT routing table to /var/www/.cache/aria2/dht.dat. 2021-11-11
08:07:57.501330 [ERROR] [DHTAutoSaveCommand.cc:112] Exception caught while saving DHT routing table to /var/www/.cache/aria2/dht.dat Exception:
[DHTRoutingTableSerializer.cc:87] errorCode=1 Failed to save DHT routing table to /var/www/.cache/aria2/dht.dat. 2021-11-11 08:37:58.477887
[ERROR] [DHTAutoSaveCommand.cc:112] Exception caught while saving DHT routing table to /var/www/.cache/aria2/dht.dat Exception:
[DHTRoutingTableSerializer.cc:87] errorCode=1 Failed to save DHT routing table to /var/www/.cache/aria2/dht.dat. 2021-11-11 09:07:59.466454
[ERROR] [DHTAutoSaveCommand.cc:112] Exception caught while saving DHT routing table to /var/www/.cache/aria2/dht.dat Exception:
[DHTRoutingTableSerializer.cc:87] errorCode=1 Failed to save DHT routing table to /var/www/.cache/aria2/dht.dat. 2021-11-11 09:38:00.446311
[ERROR] [DHTAutoSaveCommand.cc:112] Exception caught while saving DHT routing table to /var/www/.cache/aria2/dht.dat Exception:
[DHTRoutingTableSerializer.cc:87] errorCode=1 Failed to save DHT routing table to /var/www/.cache/aria2/dht.dat. 2021-11-11 10:08:01.424584
[ERROR] [DHTAutoSaveCommand.cc:112] Exception caught while saving DHT routing table to /var/www/.cache/aria2/dht.dat Exception:
[DHTRoutingTableSerializer.cc:87] errorCode=1 Failed to save DHT routing table to /var/www/.cache/aria2/dht.dat. 2021-11-11 10:38:02.415577
[ERROR] [DHTAutoSaveCommand.cc:112] Exception caught while saving DHT routing table to /var/www/.cache/aria2/dht.dat Exception:
[DHTRoutingTableSerializer.cc:87] errorCode=1 Failed to save DHT routing table to /var/www/.cache/aria2/dht.dat. 2021-11-11 11:08:03.402529
[ERROR] [DHTAutoSaveCommand.cc:112] Exception caught while saving DHT routing table to /var/www/.cache/aria2/dht.dat Exception:
[DHTRoutingTableSerializer.cc:87] errorCode=1 Failed to save DHT routing table to /var/www/.cache/aria2/dht.dat. 2021-11-11 11:38:04.382765
[ERROR] [DHTAutoSaveCommand.cc:112] Exception caught while saving DHT routing table to /var/www/.cache/aria2/dht.dat Exception:
[DHTRoutingTableSerializer.cc:87] errorCode=1 Failed to save DHT routing table to /var/www/.cache/aria2/dht.dat. 2021-11-11 12:08:05.356779
[ERROR] [DHTAutoSaveCommand.cc:112] Exception caught while saving DHT routing table to /var/www/.cache/aria2/dht.dat Exception:
[DHTRoutingTableSerializer.cc:87] errorCode=1 Failed to save DHT routing table to /var/www/.cache/aria2/dht.dat. 2021-11-11 12:38:06.325845
[ERROR] [DHTAutoSaveCommand.cc:112] Exception caught while saving DHT routing table to /var/www/.cache/aria2/dht.dat Exception:
[DHTRoutingTableSerializer.cc:87] errorCode=1 Failed to save DHT routing table to /var/www/.cache/aria2/dht.dat. 2021-11-11 13:08:07.314004
[ERROR] [DHTAutoSaveCommand.cc:112] Exception caught while saving DHT routing table to /var/www/.cache/aria2/dht.dat Exception:
[DHTRoutingTableSerializer.cc:87] errorCode=1 Failed to save DHT routing table to /var/www/.cache/aria2/dht.dat. 2021-11-11 13:38:08.293105
[ERROR] [DHTAutoSaveCommand.cc:112] Exception caught while saving DHT routing table to /var/www/.cache/aria2/dht.dat Exception:
[DHTRoutingTableSerializer.cc:87] errorCode=1 Failed to save DHT routing table to /var/www/.cache/aria2/dht.dat. 2021-11-11 14:08:09.276741
[ERROR] [DHTAutoSaveCommand.cc:112] Exception caught while saving DHT routing table to /var/www/.cache/aria2/dht.dat Exception:
[DHTRoutingTableSerializer.cc:87] errorCode=1 Failed to save DHT routing table to /var/www/.cache/aria2/dht.dat. 2021-11-11 14:38:10.250280
[ERROR] [DHTAutoSaveCommand.cc:112] Exception caught while saving DHT routing table to /var/www/.cache/aria2/dht.dat Exception:
[DHTRoutingTableSerializer.cc:87] errorCode=1 Failed to save DHT routing table to /var/www/.cache/aria2/dht.dat. 2021-11-11 15:08:11.219902
[ERROR] [DHTAutoSaveCommand.cc:112] Exception caught while saving DHT routing table to /var/www/.cache/aria2/dht.dat Exception:
[DHTRoutingTableSerializer.cc:87] errorCode=1 Failed to save DHT routing table to /var/www/.cache/aria2/dht.dat. 2021-11-11 15:38:12.200048
[ERROR] [DHTAutoSaveCommand.cc:112] Exception caught while saving DHT routing table to /var/www/.cache/aria2/dht.dat Exception:
[DHTRoutingTableSerializer.cc:87] errorCode=1 Failed to save DHT routing table to /var/www/.cache/aria2/dht.dat. 2021-11-11 16:08:13.182278
[ERROR] [DHTAutoSaveCommand.cc:112] Exception caught while saving DHT routing table to /var/www/.cache/aria2/dht.dat Exception:
[DHTRoutingTableSerializer.cc:87] errorCode=1 Failed to save DHT routing table to /var/www/.cache/aria2/dht.dat. 2021-11-11 16:38:13.216174
[ERROR] [DHTAutoSaveCommand.cc:112] Exception caught while saving DHT routing table to /var/www/.cache/aria2/dht.dat Exception:
[DHTRoutingTableSerializer.cc:87] errorCode=1 Failed to save DHT routing table to /var/www/.cache/aria2/dht.dat. 2021-11-11 16:56:07.519319
[ERROR] [RequestGroupMan.cc:572] Exception caught Exception: [RequestGroup.cc:290] errorCode=12 InfoHash
33e28a12eada81cae81b72ed1dfb69a27034a2b7 is already registered. 2021-11-11 16:56:26.146588 [ERROR] [RequestGroupMan.cc:572] Exception caught
Exception: [RequestGroup.cc:290] errorCode=12 InfoHash 33e28a12eada81cae81b72ed1dfb69a27034a2b7 is already registered. 2021-11-11
16:56:50.459304 [ERROR] [RequestGroupMan.cc:572] Exception caught Exception: [RequestGroup.cc:290] errorCode=12 InfoHash
33e28a12eada81cae81b72ed1dfb69a27034a2b7 is already registered. 2021-11-11 16:57:58.799225 [ERROR] [RequestGroupMan.cc:572] Exception caught
Exception: [RequestGroup.cc:290] errorCode=12 InfoHash 33e28a12eada81cae81b72ed1dfb69a27034a2b7 is already registered. 2021-11-11
16:59:35.428660 [ERROR] [RequestGroupMan.cc:572] Exception caught Exception: [util.cc:1948] errNum=13 errorCode=18 Failed to make the directory
/mnt/ncdata/McCainM/files/Torrents/Warhammer Novels Collection, cause: Permission denied 2021-11-11 17:08:14.076723 [ERROR]
[DHTAutoSaveCommand.cc:112] Exception caught while saving DHT routing table to /var/www/.cache/aria2/dht.dat Exception:
[DHTRoutingTableSerializer.cc:87] errorCode=1 Failed to save DHT routing table to /var/www/.cache/aria2/dht.dat. 2021-11-11 17:38:16.468530
[ERROR] [DHTAutoSaveCommand.cc:112] Exception caught while saving DHT routing table to /var/www/.cache/aria2/dht.dat Exception:
[DHTRoutingTableSerializer.cc:87] errorCode=1 Failed to save DHT routing table to /var/www/.cache/aria2/dht.dat. 2021-11-11 18:08:21.897416
[ERROR] [DHTAutoSaveCommand.cc:112] Exception caught while saving DHT routing table to /var/www/.cache/aria2/dht.dat Exception:
[DHTRoutingTableSerializer.cc:87] errorCode=1 Failed to save DHT routing table to /var/www/.cache/aria2/dht.dat. 2021-11-11 18:38:24.213814
[ERROR] [DHTAutoSaveCommand.cc:112] Exception caught while saving DHT routing table to /var/www/.cache/aria2/dht.dat Exception:
[DHTRoutingTableSerializer.cc:87] errorCode=1 Failed to save DHT routing table to /var/www/.cache/aria2/dht.dat. 2021-11-11 18:43:58.605675
[ERROR] [DHTAutoSaveCommand.cc:112] Exception caught while saving DHT routing table to /var/www/.cache/aria2/dht.dat Exception:
[DHTRoutingTableSerializer.cc:87] errorCode=1 Failed to save DHT routing table to /var/www/.cache/aria2/dht.dat. 2021-11-12 04:53:48.549809
[ERROR] [HttpListenCommand.cc:112] IPv4 RPC: failed to bind TCP port 6800 Exception: [SocketCore.cc:312] errorCode=1 Failed to bind a socket,
cause: Address already in use 2021-11-12 04:53:48.549954 [ERROR] [HttpListenCommand.cc:112] IPv6 RPC: failed to bind TCP port 6800 Exception:
[SocketCore.cc:312] errorCode=1 Failed to bind a socket, cause: Address already in use 2021-11-12 04:53:48.550039 [ERROR]
[MultiUrlRequestInfo.cc:297] Exception caught Exception: [DownloadEngineFactory.cc:219] errorCode=1 Failed to setup RPC server. 2022-02-23
03:46:57.153363 [ERROR] [HttpListenCommand.cc:112] IPv4 RPC: failed to bind TCP port 6800 Exception: [SocketCore.cc:312] errorCode=1 Failed to
bind a socket, cause: Address already in use 2022-02-23 03:46:57.158981 [ERROR] [HttpListenCommand.cc:112] IPv6 RPC: failed to bind TCP port
6800 Exception: [SocketCore.cc:312] errorCode=1 Failed to bind a socket, cause: Address already in use 2022-02-23 03:46:57.159085 [ERROR]
[MultiUrlRequestInfo.cc:297] Exception caught Exception: [DownloadEngineFactory.cc:219] errorCode=1 Failed to setup RPC server. 2022-02-23
03:50:52.188211 [ERROR] [HttpListenCommand.cc:112] IPv4 RPC: failed to bind TCP port 6800 Exception: [SocketCore.cc:312] errorCode=1 Failed to
bind a socket, cause: Address already in use 2022-02-23 03:50:52.188396 [ERROR] [HttpListenCommand.cc:112] IPv6 RPC: failed to bind TCP port
6800 Exception: [SocketCore.cc:312] errorCode=1 Failed to bind a socket, cause: Address already in use 2022-02-23 03:50:52.188512 [ERROR]
[MultiUrlRequestInfo.cc:297] Exception caught Exception: [DownloadEngineFactory.cc:219] errorCode=1 Failed to setup RPC server. 2022-02-23
04:03:57.030077 [ERROR] [AbstractCommand.cc:349] CUID#16 - Download aborted.
URI=udp://tracker.coppersurfer.tk:6969/announce?info_hash=%9D%7BN%8AT%BD%DF%CC%A6%B1%D8%0805%E3%B7%0Eod~&peer_id=A2-1-35-0-%96N%7C%B8%EC%2C%09A%DC%7D&uploaded=0&downloaded=0&left=0&compact=1&key=%7C%B8%EC%2C%09A%DC%7D&numwant=50&no_peer_id=1&port=6905&event=started&supportcrypto=1
Exception: [AbstractCommand.cc:351] errorCode=1
URI=udp://tracker.coppersurfer.tk:6969/announce?info_hash=%9D%7BN%8AT%BD%DF%CC%A6%B1%D8%0805%E3%B7%0Eod~&peer_id=A2-1-35-0-%96N%7C%B8%EC%2C%09A%DC%7D&uploaded=0&downloaded=0&left=0&compact=1&key=%7C%B8%EC%2C%09A%DC%7D&numwant=50&no_peer_id=1&port=6905&event=started&supportcrypto=1
  -> [InitiateConnectionCommandFactory.cc:87] errorCode=1 udp is not supported yet. 2022-02-23 04:03:57.030154 [ERROR] [AbstractCommand.cc:349]
CUID#17 - Download aborted.
URI=udp://tracker.coppersurfer.tk:6969/announce?info_hash=3%E2%8A%12%EA%DA%81%CA%E8%1Br%ED%1D%FBi%A2p4%A2%B7&peer_id=A2-1-35-0-%96N%7C%B8%EC%2C%09A%DC%7D&uploaded=0&downloaded=0&left=0&compact=1&key=%7C%B8%EC%2C%09A%DC%7D&numwant=50&no_peer_id=1&port=6905&event=started&supportcrypto=1
Exception: [AbstractCommand.cc:351] errorCode=1
URI=udp://tracker.coppersurfer.tk:6969/announce?info_hash=3%E2%8A%12%EA%DA%81%CA%E8%1Br%ED%1D%FBi%A2p4%A2%B7&peer_id=A2-1-35-0-%96N%7C%B8%EC%2C%09A%DC%7D&uploaded=0&downloaded=0&left=0&compact=1&key=%7C%B8%EC%2C%09A%DC%7D&numwant=50&no_peer_id=1&port=6905&event=started&supportcrypto=1
  -> [InitiateConnectionCommandFactory.cc:87] errorCode=1 udp is not supported yet. 2022-02-23 04:03:59.084096 [ERROR] [AbstractCommand.cc:349]
CUID#18 - Download aborted.
URI=udp://tracker.openbittorrent.com:6969/announce?info_hash=%9D%7BN%8AT%BD%DF%CC%A6%B1%D8%0805%E3%B7%0Eod~&peer_id=A2-1-35-0-%96N%7C%B8%EC%2C%09A%DC%7D&uploaded=0&downloaded=0&left=0&compact=1&key=%7C%B8%EC%2C%09A%DC%7D&numwant=50&no_peer_id=1&port=6905&event=started&supportcrypto=1
Exception: [AbstractCommand.cc:351] errorCode=1
URI=udp://tracker.openbittorrent.com:6969/announce?info_hash=%9D%7BN%8AT%BD%DF%CC%A6%B1%D8%0805%E3%B7%0Eod~&peer_id=A2-1-35-0-%96N%7C%B8%EC%2C%09A%DC%7D&uploaded=0&downloaded=0&left=0&compact=1&key=%7C%B8%EC%2C%09A%DC%7D&numwant=50&no_peer_id=1&port=6905&event=started&supportcrypto=1
  -> [InitiateConnectionCommandFactory.cc:87] errorCode=1 udp is not supported yet. 2022-02-23 04:03:59.084149 [ERROR] [AbstractCommand.cc:349]
CUID#19 - Download aborted.
URI=udp://tracker.openbittorrent.com:6969/announce?info_hash=3%E2%8A%12%EA%DA%81%CA%E8%1Br%ED%1D%FBi%A2p4%A2%B7&peer_id=A2-1-35-0-%96N%7C%B8%EC%2C%09A%DC%7D&uploaded=0&downloaded=0&left=0&compact=1&key=%7C%B8%EC%2C%09A%DC%7D&numwant=50&no_peer_id=1&port=6905&event=started&supportcrypto=1
Exception: [AbstractCommand.cc:351] errorCode=1
URI=udp://tracker.openbittorrent.com:6969/announce?info_hash=3%E2%8A%12%EA%DA%81%CA%E8%1Br%ED%1D%FBi%A2p4%A2%B7&peer_id=A2-1-35-0-%96N%7C%B8%EC%2C%09A%DC%7D&uploaded=0&downloaded=0&left=0&compact=1&key=%7C%B8%EC%2C%09A%DC%7D&numwant=50&no_peer_id=1&port=6905&event=started&supportcrypto=1
  -> [InitiateConnectionCommandFactory.cc:87] errorCode=1 udp is not supported yet. 2022-02-23 04:04:01.344935 [ERROR] [AbstractCommand.cc:349]
CUID#23 - Download aborted.
URI=udp://tracker.opentrackr.org:1337?info_hash=%9D%7BN%8AT%BD%DF%CC%A6%B1%D8%0805%E3%B7%0Eod~&peer_id=A2-1-35-0-%96N%7C%B8%EC%2C%09A%DC%7D&uploaded=0&downloaded=0&left=0&compact=1&key=%7C%B8%EC%2C%09A%DC%7D&numwant=50&no_peer_id=1&port=6905&event=started&supportcrypto=1
Exception: [AbstractCommand.cc:351] errorCode=1
URI=udp://tracker.opentrackr.org:1337?info_hash=%9D%7BN%8AT%BD%DF%CC%A6%B1%D8%0805%E3%B7%0Eod~&peer_id=A2-1-35-0-%96N%7C%B8%EC%2C%09A%DC%7D&uploaded=0&downloaded=0&left=0&compact=1&key=%7C%B8%EC%2C%09A%DC%7D&numwant=50&no_peer_id=1&port=6905&event=started&supportcrypto=1
  -> [InitiateConnectionCommandFactory.cc:87] errorCode=1 udp is not supported yet. 2022-02-23 04:04:01.345012 [ERROR] [AbstractCommand.cc:349]
CUID#24 - Download aborted.
URI=udp://9.rarbg.to:2710/announce?info_hash=3%E2%8A%12%EA%DA%81%CA%E8%1Br%ED%1D%FBi%A2p4%A2%B7&peer_id=A2-1-35-0-%96N%7C%B8%EC%2C%09A%DC%7D&uploaded=0&downloaded=0&left=0&compact=1&key=%7C%B8%EC%2C%09A%DC%7D&numwant=50&no_peer_id=1&port=6905&event=started&supportcrypto=1
Exception: [AbstractCommand.cc:351] errorCode=1
URI=udp://9.rarbg.to:2710/announce?info_hash=3%E2%8A%12%EA%DA%81%CA%E8%1Br%ED%1D%FBi%A2p4%A2%B7&peer_id=A2-1-35-0-%96N%7C%B8%EC%2C%09A%DC%7D&uploaded=0&downloaded=0&left=0&compact=1&key=%7C%B8%EC%2C%09A%DC%7D&numwant=50&no_peer_id=1&port=6905&event=started&supportcrypto=1
  -> [InitiateConnectionCommandFactory.cc:87] errorCode=1 udp is not supported yet. 2022-02-23 04:04:03.345558 [ERROR] [AbstractCommand.cc:349]
CUID#25 - Download aborted.
URI=udp://tracker.leechers-paradise.org:6969/announce?info_hash=%9D%7BN%8AT%BD%DF%CC%A6%B1%D8%0805%E3%B7%0Eod~&peer_id=A2-1-35-0-%96N%7C%B8%EC%2C%09A%DC%7D&uploaded=0&downloaded=0&left=0&compact=1&key=%7C%B8%EC%2C%09A%DC%7D&numwant=50&no_peer_id=1&port=6905&event=started&supportcrypto=1
Exception: [AbstractCommand.cc:351] errorCode=1
URI=udp://tracker.leechers-paradise.org:6969/announce?info_hash=%9D%7BN%8AT%BD%DF%CC%A6%B1%D8%0805%E3%B7%0Eod~&peer_id=A2-1-35-0-%96N%7C%B8%EC%2C%09A%DC%7D&uploaded=0&downloaded=0&left=0&compact=1&key=%7C%B8%EC%2C%09A%DC%7D&numwant=50&no_peer_id=1&port=6905&event=started&supportcrypto=1
  -> [InitiateConnectionCommandFactory.cc:87] errorCode=1 udp is not supported yet. 2022-02-23 04:04:03.345633 [ERROR] [AbstractCommand.cc:349]
CUID#26 - Download aborted.
URI=udp://9.rarbg.me:2780/announce?info_hash=3%E2%8A%12%EA%DA%81%CA%E8%1Br%ED%1D%FBi%A2p4%A2%B7&peer_id=A2-1-35-0-%96N%7C%B8%EC%2C%09A%DC%7D&uploaded=0&downloaded=0&left=0&compact=1&key=%7C%B8%EC%2C%09A%DC%7D&numwant=50&no_peer_id=1&port=6905&event=started&supportcrypto=1
Exception: [AbstractCommand.cc:351] errorCode=1
URI=udp://9.rarbg.me:2780/announce?info_hash=3%E2%8A%12%EA%DA%81%CA%E8%1Br%ED%1D%FBi%A2p4%A2%B7&peer_id=A2-1-35-0-%96N%7C%B8%EC%2C%09A%DC%7D&uploaded=0&downloaded=0&left=0&compact=1&key=%7C%B8%EC%2C%09A%DC%7D&numwant=50&no_peer_id=1&port=6905&event=started&supportcrypto=1
  -> [InitiateConnectionCommandFactory.cc:87] errorCode=1 udp is not supported yet.`


This is the line I ran to "configure rpc" I think I copied it from a tutorial somewhere, I can't find it now though. 

`sudo -u www-data aria2c --enable-rpc --rpc-allow-origin-all -c -D --log=/var/log/aria2c/aria2c.log --check-certificate=false --save-session=/var/local/aria2c/aria2c.sess --save-session-interval=2 --continue=true --input-file=/var/local/aria2c/aria2c.sess --rpc-save-upload-metadata=true --force-save=true --log-level=warn --rpc-secret=********** --enable-dht=false --enable-dht6=false --rpc-allow-ip=127.0.0.1`
